File: Remove Covered Intervals/Remove Covered Intervals.java
// Solution for Remove Covered Intervals in JAVA

class Solution {
    public int removeCoveredIntervals(int[][] intervals) {
        // Sort by left endpoint ascending; if equal, right endpoint descending.
        Arrays.sort(intervals, (a, b) -> {
            if (a[0] == b[0]) {
                return b[1] - a[1];
            }
            return a[0] - b[0];
        });

        int maxRight = 0;
        int remaining = 0;

        for (int[] interval : intervals) {
            if (interval[1] > maxRight) {
                maxRight = interval[1];
                remaining++;
            }
        }

        return remaining;
    }
}

File: Remove Covered Intervals/Remove Covered Intervals.py
# Solution for Remove Covered Intervals in PY

class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        # Sort by left endpoint ascending; if equal, right endpoint descending.
        intervals.sort(key=lambda x: (x[0], -x[1]))

        max_right = 0
        remaining = 0

        for left, right in intervals:
            if right > max_right:
                max_right = right
                remaining += 1

        return remaining
